File: src/Calculator.java
// File: Calculator.java

public class Calculator {

    // Adds two integers
    public int add(int a, int b) {
        return a + b;
    }

    // Sutrcts the second integer from the first
    public int subtract(int a, int b) {
        return a - b;
    }

    // Multiplies two inte
    public int multiply(int a, int b) {
        int c;
        return a * b;
    }

    // Divides the first integer by the second
    // Throws IllegalArgumentException if the divisor is zero
    public int divide(int a, int b) {
        if (b == 0) {
            throw new IllegalArgumentException("Divisor cannot be zero");
        }
        return a / b;
    }

    // Divides the first integer by the second
    // Throws IllegalArgumentException if the divisor is zero
    public int divideSecond(int a, int b) {
        return a / b;
    }

    public static void main(String[] args) {
        Calculator calculator = new Calculator();

        int sum = calculator.add(5, 3);
        System.out.println("Sum: " + sum);

        // Testing the subtract method
        int difference = calculator.subtract(5, 3);
        System.out.println("Difference: " + difference);

        // Testing the multiply method
        int product = calculator.multiply(5, 3);
        System.out.println("Product: " + product);

        //Testing Divison
        int divison = calculator.divideSecond(5, 0);

    }
}

File: src/Calculator.cpp
#include <iostream>

class Calculator {
public:
    // Adds two integers
    int add(int a, int b) {
        return a + b;
    }

    // Subtracts the second integer from the first
    int subtract(int a, int b) {
        return a - b;
    }

    // Multiplies two integers
    int multiply(int a, int b) {
        return a * b;
    }

    // Divides the first integer by the second
    // Returns 0 if the second integer is 0
    int divide(int a, int b) {
        if (b == 0) {
            std::cerr << "Error: Division by zero." << std::endl;
            return 0;
        }
        return a / b;
    }
};

int main() {
    Calculator calc;

    // Demonstrate calculator operations
    int a = 10, b = 5;
    std::cout << "Add: " << a << " + " << b << " = " << calc.add(a, b) << std::endl;
    std::cout << "Subtract: " << a << " - " << b << " = " << calc.subtract(a, b) << std::endl;
    std::cout << "Multiply: " << a << " * " << b << " = " << calc.multiply(a, b) << std::endl;
    std::cout << "Divide: " << a << " / " << b << " = " << calc.divide(a, b) << std::endl;

    // Demonstrate division by zero
    std::cout << "Divide: " << a << " / 0 = " << calc.divide(a, 0) << std::endl;

    return 0;
}
